Sources/_MatchingEngine/Regex/Parse/CharacterPropertyClassification.swift

```diff
@@ -0,0 +1,395 @@
+extension Source {
+  static private func normalizeForCharacterPropertyMatching(
+    _ str: String
+  ) -> String {
+    // This follows the rules provided by UAX44-LM3, including the dropping of
+    // an "is" prefix, which isn't required by UTS#18 RL1.2, but is nice for
+    // consistency with other engines and the Unicode.Scalar.Properties names.
+    let str = str.filter { !$0.isWhitespace && $0 != "_" && $0 != "-" }
+                 .lowercased()
+    if str.hasPrefix("is") {
+      return String(str.dropFirst(2))
```

Comment:
Do we have to check count and/or other things here? Otherwise there might be some conflict with scripts that have `IS` as a prefix.

Reply:
Good catch! There are only a couple of cases of properties beginning with `is` in the spec, and none of them affect the cases implemented so far, but I think we should probably change this to first try matching without dropping the prefix, and then try again with it dropped.